contracts/protocol/libraries/logic/PoolExtendedLogic.sol

```diff
@@ -0,0 +1,367 @@
+import {IPToken} from "../../../interfaces/IPToken.sol";
+import {IERC721} from "../../../dependencies/openzeppelin/contracts/IERC721.sol";
+import {PercentageMath} from "../../../protocol/libraries/math/PercentageMath.sol";
+
+library PoolExtendedLogic {
+    using Math for uint256;
+    using UserConfiguration for DataTypes.UserConfigurationMap;
+    using PercentageMath for uint256;
+
+    event ReserveUsedAsCollateralDisabled(
+        address indexed reserve,
+        address indexed user
+    );
+
+    event BlurExchangeRequestInitiated(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    event BlurExchangeRequestFulfilled(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    event BlurExchangeRequestRejected(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    function executeInitiateBlurExchangeRequest(
+        DataTypes.PoolStorage storage ps,
+        IPoolAddressesProvider poolAddressProvider,
+        DataTypes.BlurBuyWithCreditRequest calldata request
+    ) external {
+        //check request status
+        bytes32 requestHash = _calculateBlurExchangeRequestHash(request);
+        require(
+            ps._blurExchangeRequestStatus[requestHash] ==
+                DataTypes.BlurBuyWithCreditRequestStatus.Default,
+            Errors.INVALID_REQUEST_STATUS
+        );
+
+        address weth = poolAddressProvider.getWETH();
+        address oracle = poolAddressProvider.getPriceOracle();
+        address keeper = ps._blurExchangeKeeper;
+        DataTypes.UserConfigurationMap storage userConfig = ps._usersConfig[
+            request.initiator
+        ];
+
+        ps._blurOngoingRequestAmount += 1;
+        uint256 requestFee = request.listingPrice.percentMul(
+            ps._blurExchangeRequestFeeRate
+        );
+        ValidationLogic.validateInitiateBlurExchangeRequest(
+            ps._reserves[request.collection],
+            request,
+            ps._blurExchangeEnable,
+            keeper,
+            requestFee,
+            ps._blurOngoingRequestAmount,
+            ps._blurOngoingRequestLimit,
+            oracle
+        );
+
+        //mint nToken to release credit value
+        {
+            DataTypes.ERC721SupplyParams[]
+                memory tokenData = new DataTypes.ERC721SupplyParams[](1);
+            tokenData[0] = DataTypes.ERC721SupplyParams(request.tokenId, true);
+            SupplyLogic.executeSupplyERC721(
+                ps._reserves,
+                userConfig,
+                DataTypes.ExecuteSupplyERC721Params({
```

Comment:
this will update interest rates but we dont have real money into ptoken

Reply:
We don't update interest rates for ERC721